### PR TITLE
feature/app-5567-checkbox-group-orientation

### PR DESCRIPTION
**Pull request by acassiovilasboas**
## Description

- add checkboxGroup orientation veritcal/horizontal

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Screenshots (if appropriate):

Simulator Screenshot - iPhone 15 - 2024-03-25 at 14 11 32

## Checklist:

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) document
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have made corresponding changes to the documentation (if appropriate)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All new and existing tests passed

### CODE REVIEW

File: OceanComponents/Classes/Components SwiftUI/CheckboxGroup/SwiftUI+CheckboxGroup.swift [hunk 1]
```diff
@@ -42,15 +45,23 @@ extension OceanSwiftUI {
             public var id: String
             public var title: String
             public var isSelected: Bool
+            public var isEnable: Bool
```
Comment (renanmassaroto) [inline OceanComponents/Classes/Components SwiftUI/CheckboxGroup/SwiftUI+CheckboxGroup.swift:48]: Troca pra isEnabled

File: OceanComponents/Classes/Components SwiftUI/CheckboxGroup/SwiftUI+CheckboxGroup.swift [hunk 2]
```diff
@@ -106,9 +135,7 @@ extension OceanSwiftUI {
                 HStack(alignment: .center, spacing: Ocean.size.spacingStackXxs) {
                     ZStack {
                         RoundedRectangle(cornerRadius: 4)
-                            .fill(item.isSelected
-                                  ? Color(Ocean.color.colorComplementaryPure)
-                                  : Color(Ocean.color.colorInterfaceLightPure))
+                            .fill(item.isSelected ? (item.isEnable ? Color(Ocean.color.colorComplementaryPure) : Color(Ocean.color.colorInterfaceLightDown)) : Color(Ocean.color.colorInterfaceLightPure))
```
Comment (renanmassaroto) [inline OceanComponents/Classes/Components SwiftUI/CheckboxGroup/SwiftUI+CheckboxGroup.swift:138]: Sugiro extrair para um método pq ler um if ternário dentro de outro if ternário fica ruim, tão extenso assim eu acho que fica ainda pior.

File: OceanComponents/Classes/Extensions SwiftUI/View+Extensions.swift [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import Foundation
+import SwiftUI
+
+extension View {
+    /// Applies the given transform if the given condition evaluates to `true`.
+    /// - Parameters:
+    ///   - condition: The condition to evaluate.
+    ///   - transform: The transform to apply to the source `View`.
+    /// - Returns: Either the original `View` or the modified `View` if the condition is `true`.
+    @ViewBuilder func `if`<Content: View>(_ condition: Bool, transform: (Self) -> Content) -> some View {
```
Comment (renanmassaroto) [inline OceanComponents/Classes/Extensions SwiftUI/View+Extensions.swift:17]: Sugiro trocar para um nome que dê mais contexto ao que a extension faz, por exemplo "transformIf" ou algo do tipo

**Review by renanmassaroto — APPROVED** [review]

File: OceanComponents/Classes/Extensions SwiftUI/View+Extensions.swift (hunk 3 above)
Comment (acassiovilasboas) [reply OceanComponents/Classes/Extensions SwiftUI/View+Extensions.swift:17]: Boa, o que acha de conditionalTransform?

**Review by renanmassaroto — APPROVED** [review]

**Pull request merged**